src/new/wallet.js

```diff
@@ -1144,10 +1212,15 @@ class HathorWallet extends EventEmitter {
     // Save the transaction in the storage
     await this.storage.addTx(newTx);
 
-    // check gap-limit and load more addresses if needed
-    const fillGapLimit = await checkGapLimit(this.storage);
-    if (fillGapLimit !== null) {
-      await syncHistory(fillGapLimit.nextIndex, fillGapLimit.count, this.storage, this.conn);
+    // check address scanning policy and load more addresses if needed
+    const loadMoreAddresses = await checkScanningPolicy(this.storage);
+    if (loadMoreAddresses !== null) {
+      await syncHistory(
+        loadMoreAddresses.nextIndex,
+        loadMoreAddresses.count,
+        this.storage,
+        this.conn,
+      );
```

Comment:
Here we check if the new transaction received requires the wallet to load more addresses.
The index-limit will not load more addresses unless configured.